solr/solrj/src/java/org/apache/solr/common/util/XML.java

```diff
@@ -118,16 +122,16 @@ public final static void writeUnescapedXML(Writer out, String tag, String val, O
       out.write('>');
     } else {
       out.write('>');
-      out.write(val);
+      val.write(out);
       out.write('<');
       out.write('/');
       out.write(tag);
       out.write('>');
     }
   }
 
-  /** escapes character data in val if shouldEscape is true*/
-  public final static void writeXML(Writer out, String tag, boolean shouldEscape, String val, Object... attrs) throws IOException {
+  /** escapes character data in val */
+  public final static void writeXML(Writer out, String tag, String val, Object... attrs) throws IOException {
```

Comment:
Lets make this work in terms of the existing Writable based impl?  (less code)

Reply:
Of course, I have a commit I haven't pushed yet. waiting for ant precommit to finish running.